artemis/src/main/java/tech/pegasys/artemis/DepositCommand.java

```diff
@@ -81,44 +99,144 @@ public void generate(
+        if (!encryptKeys && !printOnStandardOut) {
+          // write keys yaml to file
+          Files.writeString(Path.of(outputFile), yamlStringBuilder.toString());
         }
       }
-      SafeFuture.allOf(futures.toArray(SafeFuture[]::new)).get(2, TimeUnit.MINUTES);
+
+      waitForTransactionReceipts(futures);
     } catch (final Throwable t) {
       STDOUT.log(
           Level.FATAL,
           "Failed to send deposit transaction: " + t.getClass() + ": " + t.getMessage());
-      System.exit(1); // Web3J creates a non-daemon thread we can't shut down. :(
+      exit(1);
     }
-    System.exit(0); // Web3J creates a non-daemon thread we can't shut down. :(
+    exit(0);
+  }
+
+  @VisibleForTesting
+  void waitForTransactionReceipts(final List<SafeFuture<TransactionReceipt>> futures)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    SafeFuture.allOf(futures.toArray(SafeFuture[]::new)).get(2, TimeUnit.MINUTES);
+  }
+
+  @VisibleForTesting
+  void exit(final int status) {
+    System.exit(status); // Web3J creates a non-daemon thread we can't shut down. :(
+  }
+
+  private String getYamlFormattedString(final Pair<BLSKeyPair, BLSKeyPair> keyPair) {
+    return String.format(
+        "- {privkey: '%s', pubkey: '%s', withdrawalPrivkey: '%s', withdrawalPubkey: '%s'}",
+        keyPair.getLeft().getSecretKey().getSecretKey().toBytes(),
+        keyPair.getLeft().getPublicKey().toBytesCompressed(),
+        keyPair.getRight().getSecretKey().getSecretKey().toBytes(),
+        keyPair.getRight().getPublicKey().toBytesCompressed());
+  }
+
+  private void generateKeystores(
+      final Pair<BLSKeyPair, BLSKeyPair> keyPair,
+      final Path keyStoreOutputDirectory,
+      final int counter) {
+    generateKeystore(keyPair.getLeft(), keyStoreOutputDirectory, "validator", counter);
+    generateKeystore(keyPair.getRight(), keyStoreOutputDirectory, "withdrawal", counter);
+  }
+
+  private void generateKeystore(
+      final BLSKeyPair key, final Path outputDir, final String prefix, final int counter) {
+    try {
+      final KdfParam kdfParam = new SCryptParam(32, Bytes32.random());
+      final Cipher cipher = new Cipher(CipherFunction.AES_128_CTR, Bytes.random(16));
+      final String randomPassword = generateRandomHexToken();
+      final KeyStoreData keyStoreData =
+          KeyStore.encrypt(
+              key.getSecretKey().getSecretKey().toBytes(),
+              generateRandomHexToken(),
+              "",
+              kdfParam,
+              cipher);
+
+      // create sub directory
+      final Path keystoreDirectory =
+          Files.createDirectories(outputDir.resolve("validator_" + counter));
+
+      // save keystore
+      KeyStoreLoader.saveToFile(
+          keystoreDirectory.resolve(prefix + "_keystore_" + counter + ".json"), keyStoreData);
+
+      // save password
+      Files.writeString(
+          keystoreDirectory.resolve(prefix + "_password_" + counter + ".txt"), randomPassword);
```

Comment:
We should have a way to specify the password to use for keystrokes.  No point writing an encrypted keystore if you write an unencrypted password for it as well.  It's probably enough to be able to specify a password for all signing keys and a (potentially different) password for all withdrawal keys.

Reply:
that make sense. Let me figure out whether to read the password from stdin or from files or from cli option.

Reply:
The password should not be included on the command line.  Using `ps auxwww` any user on the same box can see command line arguments for any running program.  It also tends to get written to places like `.bash_history`

Reply:
password input file it is ... unless we are open for "interactive shell" approach to read the input from user.

Reply:
I think an interactive mode would be an excellent option to support.  Need to check with POs as to priority and probably get Jake to think about the details.

Reply:
👍

Reply:
The interactive mode is now supported, along with reading values from environment variable as well as from a file.
~~~
--validator-password
--validator-password:env
--validator-password:file
~~~